modules/javafx.controls/src/test/java/test/javafx/scene/control/ChoiceBoxTest.java

```diff
@@ -154,6 +161,23 @@ protected void startApp(Parent root) {
         assertNull(box.getSelectionModel());
     }
 
+    @Test public void testNullSelectionModelDoesNotThrowNPEOnValueChange() {
+        ObservableList<String> items = FXCollections.observableArrayList("ITEM1", "ITEM2");
+
+        box.setSkin(new ChoiceBoxSkin<>(box));
+        box.setItems(items);
+        box.setSelectionModel(null);
+
+        try {
+            box.setValue(items.get(1));
+        } catch (Exception e) {
+            fail("ChoiceBox.setValue() should not throw an exception.");
+        }
+
```

Comment:
why the try? let the exception bubble up as-thrown (just call the formerly misbehaving method) - otherwise we are loosing information as to the origin of the error :)

Reply:
That was probably to simple to think of for me 😄  
done